client/sites/components/sites-dataviews/index.tsx

```diff
@@ -180,6 +199,17 @@ const DotcomSitesDataViews = ( {
 				enableSorting: true,
 				getValue: () => null,
 			},
+			{
+				id: 'icon',
+				label: 'Icon',
+				header: <span>{ __( 'Site' ) }</span>,
```

Comment:
Same comment here about `label`/`header`: only `label` is necessary if what you want to do is providing a text. Note that header says `Site` here.